tests/instances/eth_runner/src/post_check.rs

```diff
@@ -239,66 +301,94 @@ pub fn post_check(
+        // Check gas used
+        if res.gas_used != zk_ee::utils::u256_to_u64_saturated(&receipt.gas_used) {
+            debug!(
                     "Transaction {} has a gas mismatch: ZKsync OS used {}, reference: {}\n  Difference:{}",
                     receipt.transaction_index, res.gas_used, receipt.gas_used,
-                    zk_ee::utils::u256_to_u64_saturated(&receipt.gas_used).abs_diff(res.gas_used)
-                )
+                    gas_difference,
+                );
+            if !consistent_with_refund(res.gas_used, gas_difference) {
+                warn!("Transaction {}, gas difference should be consistent with refund\n  ZKsync OS used {}, reference: {}\n    Difference:{}",
+                    receipt.transaction_index, res.gas_used, receipt.gas_used,
+                    gas_difference,
+                );
+                // TODO: do we want this case to halt the block?
+                return Err(PostCheckError::GasMismatch {
+                    id: TxId::Index(u256_to_usize_saturated(&receipt.transaction_index)),
+                });
             }
-            // Logs check
-            assert_eq!(
-                res.logs.len(),
-                receipt.logs.len(),
+        }
+        // Logs check
+        if res.logs.len() != receipt.logs.len() {
+            error!(
                 "Transaction {} has mismatch in number of logs",
                 receipt.transaction_index
             );
-            assert!(res.logs.iter().zip(receipt.logs.iter()).all(|(l, r)| {
-                let eq = r.is_equal_to_excluding_data(l);
-                if !eq {
-                    println!("Not equal logs:\n {:#?} \nand\n {:?}", l, r)
-                }
-                if r.data.to_vec() != l.data {
-                    // We allow data to be different, as it can sometimes depend on
-                    // gas, which is not 100% equivalent (access lists)
-                    println!(
-                        "Data is not equal: we got {}, expected {}",
-                        hex::encode(l.data.clone()),
-                        hex::encode(r.data.clone())
-                    );
-                }
-
-                eq
-            }))
-        });
+            return Err(PostCheckError::IncorrectLogs {
+                id: TxId::Index(u256_to_usize_saturated(&receipt.transaction_index)),
+            });
+        }
+        for (l, r) in res.logs.iter().zip(receipt.logs.iter()) {
+            let eq = r.is_equal_to_excluding_data(l);
+            if !eq {
+                error!("Not equal logs:\n {:#?} \nand\n {:?}", l, r);
+                return Err(PostCheckError::IncorrectLogs {
+                    id: TxId::Index(u256_to_usize_saturated(&receipt.transaction_index)),
+                });
+            }
+            if r.data.to_vec() != l.data {
+                // We allow data to be different, as it can sometimes depend on
+                // gas, which is not 100% equivalent (access lists)
```

Comment:
Unclear for me. We do support access lists, right?

Reply:
Outdated comment I think, we are failing if data is different